Library/CorrelationIdManager.ts

```diff
@@ -0,0 +1,93 @@
+import https = require('https');
+import http = require('http');
+import url = require('url');
+
+class CorrelationIdManager {
+    public static correlationIdPrefix: "cid-v1:";
+
+    private static pendingLookups: {[key: string]: Function[]} = {};
+    private static completedLookups: {[key: string]: string} = {};

```

Comment:
Isn't the caller always just automatic collection code? Isn't there going to be a single config / client for automatic collection? I think you don't need static.